### PR TITLE
*: Add system variable `tidb_session_alias` to log a custom field `session_alias` in session log

### LINKED ISSUE #46071: Add a new system variable `tidb_session_alias` to print some custom trace info in log
## Feature Request

Add a new system variable `tidb_session_alias` to print some custom trace info in log

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #46071

### What is changed and how it works?

A new system variable `tidb_session_alias` with session scope is added. After this PR, we can trace some session in log identified by this field. for example:

```sql
> set @@tidb_session_alias='custom_session1';
> select * from tnoexist; -- select from a not exist table
```

Some log will be printed and you can see session_alias in it.

```
[2023/08/14 17:48:50.766 +08:00] [WARN] [session.go:2255] ["compile SQL failed"] [conn=2097154] [session_alias=custom_session1] [error="[schema:1146]Table 'test.tnoexist' doesn't exist"] [SQL="select * from tnoexist"]
[2023/08/14 17:48:50.766 +08:00] [INFO] [conn.go:1096] ["command dispatched failed"] [conn=2097154] [session_alias=custom_session1] [connInfo="id:2097154, addr:127.0.0.1:61498 status:10, collation:utf8_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="select * from tnoexist"] [txn_mode=PESSIMISTIC] [timestamp=0] [err="[schema:1146]Table 'test.tnoexist' doesn't exist"]
```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Add system variable `tidb_session_alias` to log a custom field `session_alias` in session log
```

### CODE REVIEW

**Review by CbcWestwolf — APPROVED** [review]

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -2792,6 +2792,25 @@ var defaultSysVars = []*SysVar{
 	}, GetGlobal: func(ctx context.Context, vars *SessionVars) (string, error) {
 		return BoolToOnOff(EnableCheckConstraint.Load()), nil
 	}},
+	{Scope: ScopeSession, Name: TiDBSessionAlias, Value: "", Type: TypeStr,
+		Validation: func(s *SessionVars, normalizedValue string, originalValue string, _ ScopeFlag) (string, error) {
+			if len(originalValue) > 64 {
+				s.StmtCtx.AppendWarning(ErrTruncatedWrongValue.GenWithStackByArgs(TiDBServerMemoryLimitSessMinSize, originalValue))
```
Comment (bb7133) [inline sessionctx/variable/sysvar.go:2798]: Please replace `TiDBServerMemoryLimitSessMinSize` with a`TiDBSessionAlias`.

File: sessionctx/variable/sysvar.go [hunk 2]
```diff
@@ -2792,6 +2792,25 @@ var defaultSysVars = []*SysVar{
...
+				normalizedValue = originalValue[:64]
+			}
+
+			if len(normalizedValue) > 0 && normalizedValue[len(normalizedValue)-1] == ' ' {
+				return "", ErrWrongValueForVar.GenWithStack("incorrect value for variable @@%s '%s'", TiDBSessionAlias, normalizedValue)
```
Comment (bb7133) [inline sessionctx/variable/sysvar.go:2803]:
```suggestion
				return "", ErrWrongValueForVar.GenWithStack("Incorrect value for variable @@%s '%s'", TiDBSessionAlias, normalizedValue)
```
It's better to have a upper case as the first character of an error message.

File: parser/model/model.go [hunk 3]
```diff
@@ -2152,3 +2152,11 @@ func (s WindowRepeatType) String() string {
 		return ""
 	}
 }
+
+// TraceInfo is the information for trace
+type TraceInfo struct {
+	// ConnectionID is the id of the connection
+	ConnectionID uint64 `json:"connection_id"`
+	// SessionAlias is the alias of session
+	SessionAlias string `json:"connection_alias"`
```
Comment (bb7133) [inline parser/model/model.go:2161]:
```suggestion
	SessionAlias string `json:"session_alias"`
```

File: parser/model/model.go [hunk 4]
```diff
@@ -2152,3 +2152,11 @@ func (s WindowRepeatType) String() string {
 		return ""
 	}
 }
+
+// TraceInfo is the information for trace
```
Comment (bb7133) [inline parser/model/model.go:2156]:
```suggestion
// TraceInfo is the information for trace.
```
Ditto for the rest of comments.

File: sessionctx/variable/sysvar.go [hunk 5]
```diff
@@ -2792,6 +2792,25 @@ var defaultSysVars = []*SysVar{
...
 		return BoolToOnOff(EnableCheckConstraint.Load()), nil
 	}},
+	{Scope: ScopeSession, Name: TiDBSessionAlias, Value: "", Type: TypeStr,
+		Validation: func(s *SessionVars, normalizedValue string, originalValue string, _ ScopeFlag) (string, error) {
+			if len(originalValue) > 64 {
+				s.StmtCtx.AppendWarning(ErrTruncatedWrongValue.GenWithStackByArgs(TiDBServerMemoryLimitSessMinSize, originalValue))
+				normalizedValue = originalValue[:64]
+			}
+
+			if len(normalizedValue) > 0 && normalizedValue[len(normalizedValue)-1] == ' ' {
```
Comment (bb7133) [inline sessionctx/variable/sysvar.go:2802]: IMO it's better to reuse [isIncorrectName](https://github.com/pingcap/tidb/blob/64be9ec4c0c59b7f9827b1b5b0318ade177afb9a/planner/core/preprocess.go#L1462) by moving it to some `util` package.

BTW, `[len(normalizedValue)-1] == ' '` is weird, why not just using `strings.HasSuffix(' ')`?

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**